buildenv/jenkins/omrbuild.groovy

```diff
@@ -0,0 +1,337 @@
+
+                    withEnv(SPECS[params.PLATFORM].environment) {
+                        sh 'printenv'
+                        stage('Get Sources') {
+                            if (params.ghprbPullId) {
+                                def gitConfig = scm.getUserRemoteConfigs().get(0)
+                                checkout poll: false,
+                                    scm: [$class: 'GitSCM',
+                                    branches: [[name: '${sha1}']],
+                                    userRemoteConfigs: [[name: 'origin',
+                                        refspec: "+refs/pull/${params.ghprbPullId}/merge:refs/remotes/origin/pr/${params.ghprbPullId}/merge",
+                                        url: "${gitConfig.getUrl()}"]
+                                    ]
+                                ]
+                            } else {
+                                checkout scm
+                                setBuildStatus("In Progress","PENDING","${env.GIT_COMMIT}")
+                            }
+                        }
+                        stage('Build') {
+                            if (SPECS[params.PLATFORM].ccache) {
+                                echo 'Output CCACHE stats before running and clear them'
+                                sh 'ccache -s -z'
+                            }
+
+                            for (build in SPECS[params.PLATFORM].builds) {
+                                dir("${build.buildDir}") {
+                                    echo 'Configure...'
+                                    switch (build.configure) {
+                                        case 'cmake':
+                                            sh "cmake ${build.configureArgs} .."
+                                            break
+                                        case 'make':
+                                            sh "make -f run_configure.mk OMRGLUE=./example/glue ${build.configureArgs}"
+                                            break
+                                        default:
+                                            error("Unknown Configure type")
+                                    }
+
+                                    echo 'Compile...'
+                                    sh "${build.compile}"
+                                }
+                            }
+
+                            if (SPECS[params.PLATFORM].ccache) {
+                                echo 'Output CCACHE stats after running'
+                                sh 'ccache -s'
+                            }
+                        }
+                        stage('Test') {
+                            dir('build') {
+                                if (SPECS[params.PLATFORM].test) {
+                                    echo 'Sanity Test...'
+                                    sh "ctest -V ${SPECS[params.PLATFORM].testArgs}"
```

Comment:
xlinux cmp is/was using make test in the old files. I was told it was likely missed and should be running ctest.